.github/workflows/uBench_whole_chain.yml

```diff
@@ -0,0 +1,97 @@
+name: Test the entire build and invokation chain of test examples.
+
+on: 
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  
+env:
+  GOOS: linux
+  GO111MODULE: on
+
+jobs:
+  compose_images:
+    name: Build and run chained functions serving
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v2
+    - name: go mod tidy
+      run: |
+        cd ./function-images/tests/chained-function-serving/
+        go mod tidy
+    - name: start multi-container program with docker-compose
+      run: |
+        cd ./function-images/tests/chained-function-serving/
+        docker-compose up -d
+        sleep 15s
+    - name: invoke the client
+      run: |
+        cd ./function-images/tests/chained-function-serving/
+        go run ./client/client.go
+        
+  knative_chain:
+    name: Start up and run the knative chain
+    env:
+        GITHUB_RUN_ID: ${{ github.run_id }}
+    runs-on: [self-hosted]
+    strategy:
+      fail-fast: false
+    steps:
+
+    - name: Setup TMPDIR
+      run: |
+        mkdir -p /root/tmp
+        mkdir /tmp/uBench-logs
+
+    - name: Host Info
+      run: |
+        echo $HOSTNAME
+        echo $GITHUB_RUN_ID
+
+    #- uses: actions/checkout@v2
+    #- name: setup node
+    #  run: |
+    #    ./scripts/cloudlab/setup_node.sh
+    #- name: start containerd
+    #  run: |
+    #    sudo containerd &
+    #    sleep 5s
+    #- name: start one node cluster
+    #  run : |
+    #    ./scripts/cluster/create_one_node_cluster.sh stock-only
+    #    sleep 1m
+    #- name: start consumer
+    #  run: |
+    #    kubectl create --filename ./function-images/tests/chained-function-serving/service-consumer.yaml
+    #    sleep 30s
+    #- name: start producer
+    #  run: | 
+    #    kubectl create --filename ./function-images/tests/chained-function-serving/service-producer.yaml
+    #    sleep 30s
+    #- name: show services
+    #  run: |
+    #    kn service list
+    #- name: run client
+    #  run: |
+    #    ./scripts/install_go.sh
+    #    cd ./function-images/tests/chained-function-serving/client
+    #    go build client.go
+    #    cd ..
+    #    ./client/client --addr producer.default.192.168.1.240.nip.io --pc 80 -s 3
+
+
+    - name: ls tmp logs dir
+      run: |
+        ls /tmp/uBench-logs/
```

Comment:
you don't populate the folder with files or logs. maybe that's why that error shows up